llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -2097,6 +2100,7 @@ void VPlanTransforms::addActiveLaneMask(
               [](VPUser *U) { return isa<VPWidenCanonicalIVRecipe>(U); });
   assert(FoundWidenCanonicalIVUser &&
          "Must have widened canonical IV when tail folding!");
+  VPSingleDefRecipe *HeaderMask = findHeaderMask(Plan);
```

Comment:
We need to hoist the call upwards here because once we create the VPInstruction::ActiveLaneMask the assert is violated